test/smoke/src/utils.ts

```diff
@@ -159,6 +161,61 @@ export function installCommonAfterHandlers(opts: minimist.ParsedArgs, appFn?: ()
 	});
 }
 
+// {{SQL CARBON EDIT}} - defined beforeSuite
+export function beforeSuite(opts: minimist.ParsedArgs, optionsTransform?: (opts: ApplicationOptions) => Promise<ApplicationOptions>) {
+	beforeEach(async function () {
+		const app = this.app as Application;
+
+		const name = this.currentTest!.fullTitle().replace(/[^a-z0-9\-]/ig, '_');
+		await app.code.driver.startTracing(name);
+	});
+
+	before(async function () {
+		let options: ApplicationOptions = { ...this.defaultOptions };
+
+		if (optionsTransform) {
+			options = await optionsTransform(options);
+		}
+
+		// https://github.com/microsoft/vscode/issues/34988
+		const userDataPathSuffix = [...Array(8)].map(() => Math.random().toString(36)[3]).join('');
+		const userDataDir = options.userDataDir.concat(`-${userDataPathSuffix}`);
+
+		const app = new Application({ ...options, userDataDir });
+		await app.start();
+		this.app = app;
+
+		if (opts.log) {
+			const title = this.currentTest!.fullTitle();
+			app.logger.log('*** Test start:', title);
+		}
+	});
+}
+
+// {{SQL CARBON EDIT}} - defined afterSuite
+export function afterSuite(opts: minimist.ParsedArgs) {
+	afterEach(async function () {
+		const app = this.app as Application;
+
+		const name = this.currentTest!.fullTitle().replace(/[^a-z0-9\-]/ig, '_');
+
+		if (this.currentTest?.state === 'failed' && opts.screenshots) {
+			await app.code.driver.stopTracing(name, true); // True, captures screenshots.
```

Comment:
This seems to conflict with your comment above about not being able to capture screenshots. Is this not working or something?

Reply:
That comment is from code that was carried over when I was reimplementing the missing methods.